CDCgov/prime-reportstream
# 15805 version in azure events

This PR makes the current commit hash available to the main project and adds it to Azure events.

Test Steps:
1. Run Gradle task `package` or `quickPackage` and ensure the project builds successfully.

## Changes
- Add Gradle task to generate a Version object usable in the main project.
- Add version to `ReportEventData` and supply the commit hash from the Version object.

## Checklist

### Testing

## Linked Issues
- #15805

## To be done
- #15989
